rust/src/encodings/plain.rs

```diff
@@ -84,17 +96,21 @@ impl<'a, R: Read + Seek, T: NativeType> Decoder<T> for PlainDecoder<'a, R> {
 
         let start = indices.value(0);
         let length = indices.values().last().map(|i| i - start);
-        let values = <PlainDecoder<'a, R> as Decoder<T>>::decode(self, start, &length)?;
+        let values = <PlainDecoder<'a, R, T> as Decoder>::decode(self, start, &length)?;
         let reset_indices = sub_scalar(&indices, &start);
 
         let res = take(values.as_ref(), &reset_indices);
         match res {
             Ok(arr) => Ok(arr),
-            Err(e) => Err(Error::new(ErrorKind::InvalidData, format!("Error take indices: {}", e)))
+            Err(e) => Err(Error::new(
+                ErrorKind::InvalidData,
+                format!("Error take indices: {}", e),
+            )),
         }
     }
 
-    fn value(&self, i: usize) -> Result<T> {
-        todo!()
+    fn value(&mut self, i: usize) -> Result<Box<dyn Scalar>> {
+        let arr = self._decode(i as i32, &Some(1 as i32)).unwrap();
```

Comment:
i assume this code is not ready for merge into `/rust` branch, right?

`unwrap()` here will cause panic.

Reply:
Updated to a new version that will not panic, but I still think the new code is a little redundant, any idea to polish it?